src/server/game/DataStores/DB2Stores.cpp

```diff
@@ -1922,13 +1922,14 @@ uint32 DB2Manager::GetQuestUniqueBitFlag(uint32 questId)
     return v2->UniqueBitFlag;
 }
 
-std::set<uint32> DB2Manager::GetPhasesForGroup(uint32 group) const
+std::set<uint32> const& DB2Manager::GetPhasesForGroup(uint32 group) const
 {
     auto itr = _phasesByGroup.find(group);
     if (itr != _phasesByGroup.end())
         return itr->second;
 
-    return std::set<uint32>();
+    static std::set<uint32> const empty;
```

Comment:
TODO

Reply:
Doesn't `return {}` work (still horrible)

Reply:
I kind of want want to just keep the uses intact - `for (uint32 phaseId : sDB2Manager.GetPhasesForGroup(phaseGroupId))`

Reply:
what about returning iterators

Reply:
IteratorPair? that would probably require default constructed iterators to be comparable (not all iterators work like that boost filesystem directory iterator where default constructed is .end)

Reply:
Adds boilerplate but I think that's the safest route if you don't want to go about rewriting range for loops...